### ISSUE TITLE
bug: Compiler error because of node type reference without node types

### ISSUE THREAD

**Issue by silvertech-daniel** [ionic-team/stencil#5162]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [x] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.8.2

### Current Behavior

The compiler errors during a build because of a type import from `node:child_process`, which cannot be found. This seems to be not found because the stencil project is set up to not include node types.

### Expected Behavior

Do not error the build because of a type import from node in the internal compiler implementation when the stencil project itself does not have node types.

### System Info

```shell
      System: node 19.6.0
    Platform: windows (10.0.19045)
   CPU Model: Intel(R) Core(TM) i5-7300HQ CPU @ 2.50GHz (4 cpus)
    Compiler: C:\stencil-bug\node_modules\@stencil\core\compiler\stencil.js
       Build: 1702322155
     Stencil: 4.8.2
  TypeScript: 5.2.2
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.26.0
```

### Steps to Reproduce

```sh
npm ci
# ...
npm run build
# ... ok
npm run build
# ...
[ ERROR ]  TypeScript: node_modules/@stencil/core/internal/stencil-private.d.ts:2:53
           Cannot find module 'child_process' or its corresponding type declarations.

      L1:  import { result } from './utils';
      L2:  import type { Serializable as CPSerializable } from 'child_process';
```

Happens when `lib` or `typeRoots` are set in `tsconfig.json` without node types

### Code Reproduction URL

https://github.com/silvertech-daniel/stencil-bug

**Comment by silvertech-daniel** [comment]
Looks like it was introduced here: https://github.com/ionic-team/stencil/pull/5034/files#diff-c17e8faa03832586d8dae05932c9083eb794e67b7d8ea5c7d8a8c6d7917067a7R2

**Comment by alicewriteswrongs** [comment]
hey @silvertech-daniel, thanks for reporting! just confirmed that this is a bug, taking a look at addressing it now.

**Comment by alicewriteswrongs** [comment]
Hey @silvertech-daniel, the fix for this issue was just released in [Stencil v4.9.1](https://github.com/ionic-team/stencil/releases/tag/v4.9.1)!
